### ISSUE TITLE
Strange name dependency exception while injecting the service.

### ISSUE THREAD

**Issue by wolandec** [grails/grails-core#9420]
When the service has name **LoglistService**:

```
class LoglistService {
    def get() {
        1
    }
}
```
```
class GlobalGraphController {

    def globalGraphService
    LoglistService loglistService

}
```
the BeanCreationException is appear:

```
ERROR grails.boot.GrailsApp - Application startup failed
Exception in thread "main" org.springframework.beans.factory.BeanCreationException: Error creating bean with name 'rudnaya.GlobalGraphController': Initialization of bean failed; nested exception is org.springframework.beans.ConversionNotSupportedException: Failed to convert property value of type 'LoglistService$$EnhancerBySpringCGLIB$$d0809f06' to required type 'rudnaya.LoglistService' for property 'loglistService'; nested exception is java.lang.IllegalStateException: Cannot convert value of type [LoglistService$$EnhancerBySpringCGLIB$$d0809f06] to required type [rudnaya.LoglistService] for property 'loglistService': no matching editors or conversion strategy found
	at org.springframework.beans.factory.support.AbstractAutowireCapableBeanFactory.doCreateBean(AbstractAutowireCapableBeanFactory.java:547)
	at org.springframework.beans.factory.support.AbstractAutowireCapableBeanFactory.createBean(AbstractAutowireCapableBeanFactory.java:476)
org.springframework.beans.factory.BeanCreationException: Error creating bean with name 'rudnaya.GlobalGraphController': Initialization of bean failed; nested exception is org.springframework.beans.ConversionNotSupportedException: Failed to convert property value of type 'LoglistService$$EnhancerBySpringCGLIB$$d0809f06' to required type 'rudnaya.LoglistService' for property 'loglistService'; nested exception is java.lang.IllegalStateException: Cannot convert value of type [LoglistService$$EnhancerBySpringCGLIB$$d0809f06] to required type [rudnaya.LoglistService] for property 'loglistService': no matching editors or conversion strategy found
	at org.springframework.beans.factory.support.AbstractBeanFactory$1.getObject(AbstractBeanFactory.java:303)
	at org.springframework.beans.factory.support.AbstractAutowireCapableBeanFactory.doCreateBean(AbstractAutowireCapableBeanFactory.java:547) ~[spring-beans-4.1.7.RELEASE.jar:4.1.7.RELEASE]
	at org.springframework.beans.factory.support.DefaultSingletonBeanRegistry.getSingleton(DefaultSingletonBeanRegistry.java:230)
	at org.springframework.beans.factory.support.AbstractAutowireCapableBeanFactory.createBean(AbstractAutowireCapableBeanFactory.java:476) ~[spring-beans-4.1.7.RELEASE.jar:4.1.7.RELEASE]
	at org.springframework.beans.factory.support.AbstractBeanFactory.doGetBean(AbstractBeanFactory.java:299)
	at org.springframework.beans.factory.support.AbstractBeanFactory$1.getObject(AbstractBeanFactory.java:303) ~[spring-beans-4.1.7.RELEASE.jar:4.1.7.RELEASE]
	at org.springframework.beans.factory.support.AbstractBeanFactory.getBean(AbstractBeanFactory.java:194)
	at org.springframework.beans.factory.support.DefaultListableBeanFactory.preInstantiateSingletons(DefaultListableBeanFactory.java:755)
	at org.springframework.beans.factory.support.DefaultSingletonBeanRegistry.getSingleton(DefaultSingletonBeanRegistry.java:230) ~[spring-beans-4.1.7.RELEASE.jar:4.1.7.RELEASE]
	at org.springframework.context.support.AbstractApplicationContext.finishBeanFactoryInitialization(AbstractApplicationContext.java:757)
	at org.springframework.context.support.AbstractApplicationContext.refresh(AbstractApplicationContext.java:480)
	at org.springframework.boot.context.embedded.EmbeddedWebApplicationContext.refresh(EmbeddedWebApplicationContext.java:118)
	at org.springframework.boot.SpringApplication.refresh(SpringApplication.java:687)
	at org.springframework.boot.SpringApplication.run(SpringApplication.java:321)
	at grails.boot.GrailsApp.run(GrailsApp.groovy:52)
	at org.springframework.beans.factory.support.AbstractBeanFactory.doGetBean(AbstractBeanFactory.java:299) ~[spring-beans-4.1.7.RELEASE.jar:4.1.7.RELEASE]
	at grails.boot.GrailsApp.run(GrailsApp.groovy:330)
	at org.springframework.beans.factory.support.AbstractBeanFactory.getBean(AbstractBeanFactory.java:194) ~[spring-beans-4.1.7.RELEASE.jar:4.1.7.RELEASE]
	at org.springframework.beans.factory.support.DefaultListableBeanFactory.preInstantiateSingletons(DefaultListableBeanFactory.java:755) ~[spring-beans-4.1.7.RELEASE.jar:4.1.7.RELEASE]
	at org.springframework.context.support.AbstractApplicationContext.finishBeanFactoryInitialization(AbstractApplicationContext.java:757) ~[spring-context-4.1.7.RELEASE.jar:4.1.7.RELEASE]
	at org.springframework.context.support.AbstractApplicationContext.refresh(AbstractApplicationContext.java:480) ~[spring-context-4.1.7.RELEASE.jar:4.1.7.RELEASE]
	at org.springframework.boot.context.embedded.EmbeddedWebApplicationContext.refresh(EmbeddedWebApplicationContext.java:118) ~[spring-boot-1.2.6.RELEASE.jar:1.2.6.RELEASE]
	at org.springframework.boot.SpringApplication.refresh(SpringApplication.java:687) ~[spring-boot-1.2.6.RELEASE.jar:1.2.6.RELEASE]
	at org.springframework.boot.SpringApplication.run(SpringApplication.java:321) ~[spring-boot-1.2.6.RELEASE.jar:1.2.6.RELEASE]
	at grails.boot.GrailsApp.run(GrailsApp.groovy:52) [grails-core-3.0.9.jar:3.0.9]
	at grails.boot.GrailsApp.run(GrailsApp.groovy:330) [grails-core-3.0.9.jar:3.0.9]
	at grails.boot.GrailsApp.run(GrailsApp.groovy:319) [grails-core-3.0.9.jar:3.0.9]
	at grails.boot.GrailsApp$run.call(Unknown Source) [grails-core-3.0.9.jar:3.0.9]
	at grails.boot.GrailsApp.run(GrailsApp.groovy:319)
	at org.codehaus.groovy.runtime.callsite.CallSiteArray.defaultCall(CallSiteArray.java:48) [groovy-2.4.5.jar:2.4.5]
	at grails.boot.GrailsApp$run.call(Unknown Source)
	at org.codehaus.groovy.runtime.callsite.AbstractCallSite.call(AbstractCallSite.java:113) [groovy-2.4.5.jar:2.4.5]
	at org.codehaus.groovy.runtime.callsite.CallSiteArray.defaultCall(CallSiteArray.java:48)
	at org.codehaus.groovy.runtime.callsite.AbstractCallSite.call(AbstractCallSite.java:133) [groovy-2.4.5.jar:2.4.5]
	at org.codehaus.groovy.runtime.callsite.AbstractCallSite.call(AbstractCallSite.java:113)
	at rudnaya.Application.main(Application.groovy:12) [main/:na]
	at org.codehaus.groovy.runtime.callsite.AbstractCallSite.call(AbstractCallSite.java:133)
Caused by: org.springframework.beans.ConversionNotSupportedException: Failed to convert property value of type 'LoglistService$$EnhancerBySpringCGLIB$$d0809f06' to required type 'rudnaya.LoglistService' for property 'loglistService'; nested exception is java.lang.IllegalStateException: Cannot convert value of type [LoglistService$$EnhancerBySpringCGLIB$$d0809f06] to required type [rudnaya.LoglistService] for property 'loglistService': no matching editors or conversion strategy found
	at rudnaya.Application.main(Application.groovy:12)
	at org.springframework.beans.BeanWrapperImpl.convertIfNecessary(BeanWrapperImpl.java:476) ~[spring-beans-4.1.7.RELEASE.jar:4.1.7.RELEASE]
Caused by: org.springframework.beans.ConversionNotSupportedException: Failed to convert property value of type 'LoglistService$$EnhancerBySpringCGLIB$$d0809f06' to required type 'rudnaya.LoglistService' for property 'loglistService'; nested exception is java.lang.IllegalStateException: Cannot convert value of type [LoglistService$$EnhancerBySpringCGLIB$$d0809f06] to required type [rudnaya.LoglistService] for property 'loglistService': no matching editors or conversion strategy found
	at org.springframework.beans.BeanWrapperImpl.convertForProperty(BeanWrapperImpl.java:512) ~[spring-beans-4.1.7.RELEASE.jar:4.1.7.RELEASE]
	at org.springframework.beans.BeanWrapperImpl.convertIfNecessary(BeanWrapperImpl.java:476)
	at org.springframework.beans.BeanWrapperImpl.convertForProperty(BeanWrapperImpl.java:506) ~[spring-beans-4.1.7.RELEASE.jar:4.1.7.RELEASE]
	at org.springframework.beans.BeanWrapperImpl.convertForProperty(BeanWrapperImpl.java:512)
	at org.springframework.beans.factory.support.AbstractAutowireCapableBeanFactory.convertForProperty(AbstractAutowireCapableBeanFactory.java:1523) ~[spring-beans-4.1.7.RELEASE.jar:4.1.7.RELEASE]
	at org.springframework.beans.BeanWrapperImpl.convertForProperty(BeanWrapperImpl.java:506)
	at org.springframework.beans.factory.support.AbstractAutowireCapableBeanFactory.applyPropertyValues(AbstractAutowireCapableBeanFactory.java:1482) ~[spring-beans-4.1.7.RELEASE.jar:4.1.7.RELEASE]
	at org.springframework.beans.factory.support.AbstractAutowireCapableBeanFactory.convertForProperty(AbstractAutowireCapableBeanFactory.java:1523)
	at org.springframework.beans.factory.support.AbstractAutowireCapableBeanFactory.populateBean(AbstractAutowireCapableBeanFactory.java:1222) ~[spring-beans-4.1.7.RELEASE.jar:4.1.7.RELEASE]
	at org.springframework.beans.factory.support.AbstractAutowireCapableBeanFactory.applyPropertyValues(AbstractAutowireCapableBeanFactory.java:1482)
	at org.springframework.beans.factory.support.AbstractAutowireCapableBeanFactory.doCreateBean(AbstractAutowireCapableBeanFactory.java:537) ~[spring-beans-4.1.7.RELEASE.jar:4.1.7.RELEASE]
	at org.springframework.beans.factory.support.AbstractAutowireCapableBeanFactory.populateBean(AbstractAutowireCapableBeanFactory.java:1222)
	... 19 common frames omitted
	at org.springframework.beans.factory.support.AbstractAutowireCapableBeanFactory.doCreateBean(AbstractAutowireCapableBeanFactory.java:537)
Caused by: java.lang.IllegalStateException: Cannot convert value of type [LoglistService$$EnhancerBySpringCGLIB$$d0809f06] to required type [rudnaya.LoglistService] for property 'loglistService': no matching editors or conversion strategy found
	... 19 more
Caused by: java.lang.IllegalStateException: Cannot convert value of type [LoglistService$$EnhancerBySpringCGLIB$$d0809f06] to required type [rudnaya.LoglistService] for property 'loglistService': no matching editors or conversion strategy found
	at org.springframework.beans.TypeConverterDelegate.convertIfNecessary(TypeConverterDelegate.java:287)
	at org.springframework.beans.BeanWrapperImpl.convertIfNecessary(BeanWrapperImpl.java:461)
	... 25 more
	at org.springframework.beans.TypeConverterDelegate.convertIfNecessary(TypeConverterDelegate.java:287) ~[spring-beans-4.1.7.RELEASE.jar:4.1.7.RELEASE]
	at org.springframework.beans.BeanWrapperImpl.convertIfNecessary(BeanWrapperImpl.java:461) ~[spring-beans-4.1.7.RELEASE.jar:4.1.7.RELEASE]
	... 25 common frames omitted
```

But if I change name of ~~LoglistService~~ to **LogService**, for example, everything is go fine.

```
class LogService {
    def get() {
        '1'
    }
}
```
```
class GlobalGraphController {

    def globalGraphService
    LogService logService
}
```

**Comment by graemerocher** [comment]
fixed in 3.1 which disables spring proxies

**Comment by hunterbdb** [comment]
I'm stuck with v2.5.1. What should I do if I'm encountering this issue? I have to be able to inject services into controllers but I'm getting this error because the service gets cg-lib-processed and thus does not match the declare type of the injected service field in the controller.
